cranelift/codegen/src/isa/riscv64/inst/emit.rs

```diff
@@ -3331,136 +3325,49 @@ fn emit_return_call_common_sequence(
     sink: &mut MachBuffer<Inst>,
     emit_info: &EmitInfo,
     state: &mut EmitState,
-    new_stack_arg_size: u32,
-    old_stack_arg_size: u32,
+    info: &ReturnCallInfo,
 ) {
-    // We are emitting a dynamic number of instructions and might need an
-    // island. We emit four instructions regardless of how many stack arguments
-    // we have, up to two instructions for the actual call, and then two
-    // instructions per word of stack argument space.
-    let new_stack_words = new_stack_arg_size / 8;
-    let insts = 4 + 2 + 2 * new_stack_words;
-    let space_needed = insts * u32::try_from(Inst::UNCOMPRESSED_INSTRUCTION_SIZE).unwrap();
-    if sink.island_needed(space_needed) {
```

Comment:
Is this island check now done in some common code, or do we no longer need it? I remember having some issues with missing islands when we enabled compressed jumps since with the compressed extensions we have a very short jump range.

Reply:
Thanks for catching this, that check isn't being done in any generic code. I'll add this back in!

Reply:
On further inspection, there is generic code in `VCode::emit` that emits islands based on the backend's `Inst::worst_case_size()` estimate. Both `gen_epilogue` and these `ReturnCall`/`ReturnCallInd` pseudo-instructions need to fit within that worst-case size. These pseudo-instructions might be slightly longer than `gen_epilogue` right now but they're almost the same, so I think they don't need an explicit island check.

Reply:
I did some more refactoring, and the resulting code generated for a `return_call` or `return_call_indirect` should now be pretty close to the same size as `gen_epilogue`. Lots of chatting with @jameysharp has me convinced that we shouldn't need to check for island emission at this point anymore, and I've removed the manual resets at the call-sites for `emit_return_call_common_sequence` to ensure that we trigger an assert if this isn't true.